test/e2e/README.md

```diff
@@ -0,0 +1,131 @@
+
+
+
+# E2E testing
+[![E2E Docker](https://github.com/input-output-hk/cardano-wallet/actions/workflows/e2e-docker.yml/badge.svg)](https://github.com/input-output-hk/cardano-wallet/actions/workflows/e2e-docker.yml) [![E2E Linux](https://github.com/input-output-hk/cardano-wallet/actions/workflows/e2e-linux.yml/badge.svg)](https://github.com/input-output-hk/cardano-wallet/actions/workflows/e2e-linux.yml) [![E2E MacOS](https://github.com/input-output-hk/cardano-wallet/actions/workflows/e2e-macos.yml/badge.svg)](https://github.com/input-output-hk/cardano-wallet/actions/workflows/e2e-macos.yml) [![E2E Windows](https://github.com/input-output-hk/cardano-wallet/actions/workflows/e2e-windows.yml/badge.svg)](https://github.com/input-output-hk/cardano-wallet/actions/workflows/e2e-windows.yml)
+
+E2E functional tests of cardano-wallet are running nightly on [cardano testnet](https://testnets.cardano.org/en/cardano/overview/). Running tests against public testnet allows to excercise cardano-wallet on environment close to production (mainnet) utilizing and integrating maximally all components of the Cardano ecosystem like Stake pools, SMASH, metadata token server etc.
+
+
+## Running tests
+
+In order to run tests one needs to [have ruby](https://www.ruby-lang.org/en/documentation/installation/) (at least 2.7.1).
+
+### Configuring test project
+1. Get necessary _gems_.
+```
+cd test/e2e
+bundle install
+```
+2. Decrypt `fixture_wallets.json.gpg` containing mnemonics of testnet fixture wallets using `$TESTS_E2E_FIXTURES` secret.
+
+```
+export TESTS_E2E_FIXTURES=*******
+rake fixture_wallets_decode
+```
+> :information_source:  **_TESTS_E2E_FIXTURES_** secret  is defined on https://github.com/input-output-hk/cardano-wallet/settings/secrets and also used by GH actions. Note that this step is also executed on very first test run.
+>
+### Running all tests
+In order to run all `e2e` tests one can simply run single [rake](https://github.com/ruby/rake) task:
+```ruby
+$ rake run_on[testnet]
+```
+This master task is performing also all the necessary configuration steps (i.e. getting latest testnet configs and wallet/node binaries from [Hydra](https://hydra.iohk.io/jobset/Cardano/cardano-wallet#tabs-jobs), starting everything up). All steps can also be executed as separate tasks , i.e.:
+
+```ruby
+$ rake fixture_wallets_decode
+$ rake get_latest_bins
+$ rake get_latest_configs[testnet]
+$ rake start_node_and_wallet
+$ rake wait_until_node_synced
+$ rake spec
+$ rake stop_node_and_wallet
+```
+
+##### Running tests against local wallet
+
+One can also run tests against `cardano-wallet` and `cardano-node` which are specified on machine's `$PATH`:
+
+```
+$ rake run_on[testnet,local]
```

Comment:
This way of setting the option seems odd.
Could it be done through an environment variable? Perhaps if `TESTS_E2E_BINDIR` is defined but empty, then use exes on the PATH.

Reply:
> Perhaps if TESTS_E2E_BINDIR is defined but empty, then use exes on the PATH.

I was thinking about this way too, but I thought that it's more convenient (i.e. less writing) if someone want's to switch between running tests against latest Hydra builds to testing against local installation and vice versa.

I understand that it might appear odd, but on the other hand it's sort of conventional way of providing parameters to `rake` task.

>I pushed a few changes so that it's possible to use the existing nix-build to get cardano-wallet, cardano-node, and configs, rather than downloading the latest from hydra.

Cool! how does one do that? Could you share some commands?

Reply:
> Please go ahead and merge when you're happy.

Ok, thanks. I'd like to merge today but I'd like to figure out the failures first, not to merge flaky tests.
I was thinking to run those tests on nightly basis for the time being (so not to be started on every merge) the reason is they take ~1h (and even more on Windows). Majority of the time is spent for syncing wallets. I think when we'll be able to improve restoration times (mainly ADP-690 and ADP-785) then the test execution time will benefit too.

Reply:
>Perhaps if TESTS_E2E_BINDIR is defined but empty, then use exes on the PATH.

Ok, I've added this way too. It was just modifying one line in code.

```
$ TESTS_E2E_BINDIR="" rake run_on[testnet]
```
will also use wallet/node from $PATH.